java-client/src/main/java/org/opensearch/client/opensearch/core/bulk/UpdateOperation.java

```diff
@@ -141,15 +133,11 @@ protected void serializeInternal(JsonGenerator generator, JsonpMapper mapper) {
 	public static class Builder<TDocument> extends BulkOperationBase.AbstractBuilder<Builder<TDocument>>
 			implements
 				ObjectBuilder<UpdateOperation<TDocument>> {
-		private TDocument document;
 
-		/**
-		 * Required - API name: {@code document}
-		 */
-		public final Builder<TDocument> document(TDocument value) {
-			this.document = value;
-			return this;
-		}
+        private UpdateOperationData<TDocument> data;
+
+        @Nullable
+		private TDocument document;
```

Comment:
Wouldn't document be required as non null?

Reply:
No, the document or script could be specified, so it is nullable indeed